addons/web/static/lib/qweb/qweb2.js

```diff
@@ -822,18 +824,29 @@ QWeb2.Element = (function() {
                     this.bottom("})(dict);");
                     this.indent();
                     this.top("var r = [];");
-                    this.bottom("return r.join('');");
+                    this.bottom("return context.engine.tools.markup(r.join(''));");
                 }
             }
         },
         compile_action_esc : function(value) {
+            return this.compile_action_out(value);
```

Comment:
should we deprecate t-esc as well?

Reply:
There is little reason to in the short term, on the Python side t-raw is hard-reprecated (triggers a warning and runbot will fail if it hits one), but t-esc is only "not recommended" for now: there are literally thousands of `t-esc` in the codebase so it'd be really annoying to convert them for no value at all. That can be done at a later date.

Reply:
fair enough